server/datastore/mysql/microsoft_mdm.go

```diff
@@ -90,3 +91,165 @@ func (ds *Datastore) MDMWindowsDeleteEnrolledDevice(ctx context.Context, mdmDevi
 
 	return ctxerr.Wrap(ctx, notFound("MDMWindowsEnrolledDevice"))
 }
+
+// MDMWindowsInsertPendingCommand inserts a new WindowMDMPendingCommand in the database
+func (ds *Datastore) MDMWindowsInsertPendingCommand(ctx context.Context, cmd *fleet.MDMWindowsPendingCommand) error {
+	stmt := `
+		INSERT INTO windows_mdm_pending_commands (
+		command_uuid,
+		device_id,
+		cmd_verb,
+		setting_uri,
+		setting_value,
+		data_type,
+		system_origin ) VALUES (?, ?, ?, ?, ?, ?, ?)
+	`
+	_, err := ds.writer(ctx).ExecContext(
+		ctx,
+		stmt,
+		cmd.CommandUUID,
+		cmd.DeviceID,
+		cmd.CmdVerb,
+		cmd.SettingURI,
+		cmd.SettingValue,
+		cmd.DataType,
+		cmd.SystemOrigin)
+	if err != nil {
+		if isDuplicate(err) {
+			return ctxerr.Wrap(ctx, alreadyExists("MDMWindowsPendingCommand", cmd.CommandUUID))
+		}
+		return ctxerr.Wrap(ctx, err, "inserting MDMWindowsPendingCommand")
+	}
+
+	return nil
+}
+
+// MDMWindowsListPendingCommands retrieves all commands for a given device ID from the windows_mdm_pending_commands table
+func (ds *Datastore) MDMWindowsListPendingCommands(ctx context.Context, deviceID string) ([]*fleet.MDMWindowsPendingCommand, error) {
+	var commands []*fleet.MDMWindowsPendingCommand
+
+	query := `
+        SELECT
+            command_uuid,
+            device_id,
+            cmd_verb,
+            setting_uri,
+            setting_value,
+            data_type,
+			system_origin,
+            created_at,
+            updated_at
+        FROM
+            windows_mdm_pending_commands
+        WHERE
+            device_id = ?
+    `
+
+	if err := sqlx.SelectContext(ctx, ds.reader(ctx), &commands, query, deviceID); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, nil
+		}
+
+		return nil, ctxerr.Wrap(ctx, err, "get pending Windows MDM commands by device id")
+	}
+
+	return commands, nil
+}
+
+// MDMWindowsInsertCommand inserts a new WindowMDMCommand in the database
+func (ds *Datastore) MDMWindowsInsertCommand(ctx context.Context, cmd *fleet.MDMWindowsCommand) error {
```

Comment:
What is the distinction between this method and `MDMWindowsInsertPendingCommand` above?

Reply:
That's a good question. I should have added to the PR description. 

The idea is that `MDMWindowsInsertPendingCommand ` is the table that contains the commands that were enqueued from the UI/fleetctl. These commands just contain the MDM protocol operations and do not know the final MessageID, SessionID, and CmdID values to use, as they are known only when the actual reply is sent